### PR TITLE
split agent / analyzer and api auth

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -278,7 +289,7 @@ func GetConfig() *viper.Viper {
 	return cfg
 }
 
-// SetDefault set default configuration key the value
+// SetDefault set defaults configuration key the value
```
Comment (lebauce) [inline config/config.go]: SetDefault set the default configuration value for a key ?

File: http/server.go [hunk 2]
```diff
@@ -162,15 +161,15 @@ func (s *Server) Serve() {
 		if err == http.ErrServerClosed {
 			return
 		}
-		logging.GetLogger().Errorf("Failed to Serve on %s:%d: %s", s.Addr, s.Port, err.Error())
+		logging.GetLogger().Errorf("Failed to Serve on %s:%d: %s", s.Addr, s.Port, err)
```
Comment (lebauce) [inline http/server.go]: nit: s/Server/serve/

File: http/server.go [hunk 3]
```diff
@@ -247,24 +247,33 @@ func (s *Server) serveLogin(w http.ResponseWriter, r *http.Request) {
...
-				})
+					w.WriteHeader(http.StatusOK)
 
-				w.WriteHeader(http.StatusOK)
-			} else {
-				unauthorized(w, r)
+					rbac.AddRoleForUser(login, authBackend.DefaultUserRole(login))
+					roles := rbac.GetUserRoles(login)
+
+					logging.GetLogger().Infof("User %s authicated with %s backend with roles %s", login, authBackend.Name(), roles)
```
Comment (lebauce) [inline http/server.go]: nit: s/authicated/authenticated/

File: rbac/enforcer.go [hunk 4]
```diff
@@ -110,6 +110,7 @@ func Init(kapi etcd.KeysAPI) error {
 		return err
 	}
 	loadConfigPolicy(model)
+	casbinEnforcer.BuildRoleLinks()
```
Comment (lebauce) [inline rbac/enforcer.go]: Is it required ?

Comment (safchain) [reply rbac/enforcer.go]: yes otherwise panic do to not having the RoleManager set

**Comment by safchain** [comment]
run skydive-cdd-overview-tests

**Comment by safchain** [comment]
run vagrant-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run python-tests

**Comment by safchain** [comment]
run vagrant-tests

**Comment by safchain** [comment]
rerun vagrant-tests

**Comment by safchain** [comment]
rerun vagrant-tests

**Comment by safchain** [comment]
rerun vagrant-tests

**Comment by safchain** [comment]
rerun functional-tests-backend-orientdb

**Comment by safchain** [comment]
rerun vagrant-tests

**Comment by safchain** [comment]
rerun vagrant-tests

**Comment by safchain** [comment]
rerun vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run skydive-tripleo-tests

**Comment by safchain** [comment]
run skydive-tripleo-tests

File: config/config.go [hunk 5]
```diff
@@ -44,10 +44,17 @@ var ErrNoAnalyzerSpecified = errors.New("No analyzer specified in the configurat
 var (
 	cfg           *viper.Viper
 	relocationMap = map[string][]string{
-		"analyzer.flow.max_buffer_size": {"analyzer.storage.max_flow_buffer_size"},
-		"analyzer.topology.backend":     {"graph.backend"},
-		"analyzer.flow.backend":         {"analyzer.storage.backend"},
-		"agent.capture.stats_update":    {"agent.flow.stats_update"},
+		"analyzer.flow.max_buffer_size":  {"analyzer.storage.max_flow_buffer_size"},
```
Comment (lebauce) [inline config/config.go]: Can you please use the alphabetical order ?

**Comment by safchain** [comment]
run skydive-tripleo-tests

**Comment by safchain** [comment]
run skydive-unit-tests

File: http/auth.go [hunk 6]
```diff
@@ -24,139 +24,155 @@ package http
...
 	}
-	headers.Set("Cookie", b.String())
-}
 
-func cookies(c *AuthenticationClient) []*http.Cookie {
-	var cookies []*http.Cookie
-	cookies = append(cookies, configCookies()...)
-	if c != nil {
-		cookies = append(cookies, c.AuthCookies()...)
+	if authOpts.Username != "" {
```
Comment (lebauce) [inline http/auth.go]: else if ?

**Comment by safchain** [comment]
run skydive-tripleo-tests

**Comment by safchain** [comment]
run vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-k8s-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-unit-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by safchain** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by safchain** [comment]
run skydive-cdd-overview-tests

**Comment by safchain** [comment]
run skydive-cdd-overview-tests

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
rerun tripleo-tests

**Comment by safchain** [comment]
rerun tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run tripleo-tests

**Comment by safchain** [comment]
run functional-tests-backend-orientdb

**Comment by safchain** [comment]
run vagrant-tests

**Comment by safchain** [comment]
run vagrant-tests

**Pull request merged**